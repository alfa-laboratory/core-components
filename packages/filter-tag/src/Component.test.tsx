import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { FilterTag } from './index';

describe('Snapshots tests', () => {
    it('should match snapshot', () => {
        expect(render(<FilterTag>Press me</FilterTag>)).toMatchSnapshot();
    });

    it('should match snapshot variant=alt', () => {
        expect(render(<FilterTag variant='alt'>Press me</FilterTag>)).toMatchSnapshot();
    });

    it('should match without children snapshot', () => {
        expect(render(<FilterTag />)).toMatchSnapshot();
    });

    it('should match snapshot with alt variant ', () => {
        expect(render(<FilterTag variant='alt'>Press me</FilterTag>)).toMatchSnapshot();
    });
});

describe('Classes tests', () => {
    it('should set custom class', () => {
        const className = 'custom-class';

        const { container } = render(<FilterTag className={className} />);

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set size `s` as default size', () => {
        const { container } = render(<FilterTag />);

        expect(container.firstElementChild).toHaveClass('s');
    });

    it('should set size', () => {
        const { container } = render(<FilterTag size='xs' />);

        expect(container.firstElementChild).toHaveClass('xs');
    });

    it('should set `checked` class if prop `checked` is present', () => {
        const { container } = render(<FilterTag checked={true} />);

        expect(container.firstElementChild).toHaveClass('checked');
    });

    it('should set `open` class if prop `open` is present', () => {
        const { container } = render(<FilterTag open={true} />);

        expect(container.firstElementChild).toHaveClass('open');
    });

    it('should set `alt` class if prop `variant=alt`', () => {
        const { container } = render(<FilterTag variant='alt' />);

        expect(container.firstElementChild).toHaveClass('alt');
    });

    it('should set `disabled` class if prop `disabled` is present`', () => {
        const { container } = render(<FilterTag disabled={true} />);

        expect(container.firstElementChild).toHaveClass('disabled');
    });
});

describe('Attributes tests', () => {
    it('should set disabled attribute', () => {
        const { getByRole } = render(<FilterTag disabled={true} />);
        const buttonEl = getByRole('button');
        expect(buttonEl).toBeDisabled();
    });

    it('should set data-test-id attribute', () => {
        const dataTestId = 'tag-test-id';

        const { container } = render(<FilterTag disabled={true} dataTestId={dataTestId} />);

        const testIdAttr = container.firstElementChild?.getAttribute('data-test-id');

        expect(container.firstElementChild?.getAttribute('data-test-id')).toBe(testIdAttr);
    });
});

describe('Render tests', () => {
    test('should unmount without errors', () => {
        const { unmount } = render(<FilterTag>FilterTag</FilterTag>);

        expect(unmount).not.toThrowError();
    });

    test('should contain children', () => {
        const text = 'filter Tag text';

        const { container, getByText } = render(<FilterTag>{text}</FilterTag>);

        expect(container.firstElementChild).toContainElement(getByText(text));
    });
});

describe('Interaction tests', () => {
    test('should call `onClick` prop, if filterTag not disabled', () => {
        const cb = jest.fn();
        const text = 'Press me!';
        const { getByText } = render(<FilterTag onClick={cb}>{text}</FilterTag>);

        const button = getByText(text);

        if (button) {
            fireEvent.click(button);
        }

        expect(cb).toBeCalledTimes(1);
    });

    test('should call `onClear` prop, if filterTag checked and not disabled', () => {
        const cb = jest.fn();
        const text = 'Press me!';
        const { getAllByRole } = render(
            <FilterTag checked={true} onClear={cb}>
                {text}
            </FilterTag>,
        );

        const buttons = getAllByRole('button');

        if (buttons[1]) {
            fireEvent.click(buttons[1]);
        }

        expect(cb).toBeCalledTimes(1);
    });

    test('should not call `onClick` prop, if tag is disabled', () => {
        const cb = jest.fn();
        const text = 'Press me!';

        const { getByText } = render(
            <FilterTag onClick={cb} disabled={true}>
                Press me!
            </FilterTag>,
        );
        const button = getByText(text);
        if (button) {
            fireEvent.click(button);
        }
        expect(cb).toBeCalledTimes(0);
    });

    test('should not call `onClick` prop, if tag is disabled and checked', () => {
        const cb = jest.fn();
        const text = 'Press me!';
        const { getByText } = render(
            <FilterTag onClick={cb} disabled={true} checked={true}>
                {text}
            </FilterTag>,
        );
        const button = getByText(text);
        if (button) {
            fireEvent.click(button);
        }

        expect(cb).toBeCalledTimes(0);
    });
});
