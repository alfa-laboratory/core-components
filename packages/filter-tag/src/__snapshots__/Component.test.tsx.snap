// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterTag default s"
      >
        <button
          class="valueButton s"
          type="button"
        >
          <span>
            Press me
          </span>
          <span
            class="chevron"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterTag default s"
    >
      <button
        class="valueButton s"
        type="button"
      >
        <span>
          Press me
        </span>
        <span
          class="chevron"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshots tests should match snapshot variant=alt 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterTag alt s"
      >
        <button
          class="valueButton s"
          type="button"
        >
          <span>
            Press me
          </span>
          <span
            class="chevron"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterTag alt s"
    >
      <button
        class="valueButton s"
        type="button"
      >
        <span>
          Press me
        </span>
        <span
          class="chevron"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshots tests should match snapshot with alt variant  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterTag alt s"
      >
        <button
          class="valueButton s"
          type="button"
        >
          <span>
            Press me
          </span>
          <span
            class="chevron"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterTag alt s"
    >
      <button
        class="valueButton s"
        type="button"
      >
        <span>
          Press me
        </span>
        <span
          class="chevron"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshots tests should match without children snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterTag default s"
      >
        <button
          class="valueButton s"
          type="button"
        >
          <span />
          <span
            class="chevron"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterTag default s"
    >
      <button
        class="valueButton s"
        type="button"
      >
        <span />
        <span
          class="chevron"
        >
          <svg
            fill="currentColor"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M12 14L6.5 8.5 5 10l7 7 7-7-1.5-1.5L12 14z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
